### PR TITLE
lib/api: Add /rest/status health-check (#8430) 

### PR DESCRIPTION
**Pull request by er-pa**
### Purpose

Added a status page to the REST API which can act as a health-check, this allows third-party services and devices to check the status of Syncthing without having to use an API-key or plain-texted credentials.

To accomplish this two things were adjusted:
- Exception for /rest/status to the CSRF-token-requirement
- Exception for /rest/status to the otherwise required authentication

And obviously the /rest/status functionality was added, which is a simple return of a JSON object. Since, if you reach that point - it all goes well anyways. If not, you'll get another kind of error before reaching this point.

### Testing

Two parts:
1.) Functional; visit the url and call the url in several ways, browser/curl/wget/...
2.) Security: Since we made an exception in the security-layer; hack-and-slash the URL a little to see if we don't unexpectedly reach other pages from this status-page while being unauthenticated. 

### Documentation

The API documentation will have to be updated to cover this addition.

### CODE REVIEW

File: lib/api/api.go [hunk 1]
```diff
@@ -260,6 +260,7 @@ func (s *service) Serve(ctx context.Context) error {
 	restMux.HandlerFunc(http.MethodGet, "/rest/events/disk", s.getDiskEvents)                 // [since] [limit] [timeout]
 	restMux.HandlerFunc(http.MethodGet, "/rest/stats/device", s.getDeviceStats)               // -
 	restMux.HandlerFunc(http.MethodGet, "/rest/stats/folder", s.getFolderStats)               // -
+	restMux.HandlerFunc(http.MethodGet, "/rest/status", s.getStatus)                          // -
```
[the remark this answers is not in the document]
Comment (er-pa) [reply lib/api/api.go:263]: The noauth approach is perhaps the most clear, good idea.

**Review by calmh — APPROVED** [review]

**Comment by acolomb** [comment]
Please do provide an appropriate documentation PR for the API endpoint.  We put in quite some work to get caught up and have them all documented, let's try to keep it complete.

**Comment by er-pa** [comment]
> Please do provide an appropriate documentation PR for the API endpoint.  We put in quite some work to get caught up and have them all documented, let's try to keep it complete.

I will adjust the documentation and file a PR accordingly tomorrow.

**Comment by bt90** [comment]
Our docker healthcheck is now using this endpoint #8640

**Pull request merged**